litestar/middleware/logging.py

```diff
@@ -196,8 +199,8 @@ def extract_response_data(self, scope: Scope) -> dict[str, Any]:
         serializer = get_serializer_from_scope(scope)
         extracted_data = self.response_extractor(
             messages=(
-                get_litestar_scope_state(scope, HTTP_RESPONSE_START, pop=True),
-                get_litestar_scope_state(scope, HTTP_RESPONSE_BODY, pop=True),
+                pop_litestar_scope_state(scope, SCOPE_STATE_HTTP_RESPONSE_START_KEY),
+                pop_litestar_scope_state(scope, SCOPE_STATE_HTTP_RESPONSE_BODY_KEY),
```

Comment:
This was a type error because the old `pop=True` behavior would automatically default to return `None` if the keys don't exist, but the response extractor is not typed to receive optional messages.

This new function behaves the same as `dict.pop()` so these cannot be `None` if this doesn't raise.